### PR TITLE
Added nix flake and nix shell

### PR DESCRIPTION
**Pull request by AusCyberman**
## Description
Added a nix flake with defaultPackage and app, added default.nix for non flake systems. also added a shell.nix that works on flake and non flake systems

## Additional Notes

Works perfectly on my system

### CODE REVIEW

File: shell.nix [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    rust-analyzer
+    gcc
+    gtk3
+    pkg-config
+    rustfmt
+    clippy
+  ];
+
+
+  RUST_SRC_PATH = "${pkgs.rust.packages.stable.rustPlatform.rustLibSrc}";
```
Comment (legendofmiracles) [inline shell.nix:22]: This points to the normal rustc that's packaged in nixpkgs, right?

Can this be replaced with the one that fenix provides?

**Review by legendofmiracles — COMMENTED** [review]
There are a few trailing newlines that should be removed, and deno is a development dependency for generating the documentation

File: shell.nix (hunk 1 above)
Comment (AusCyberman) [reply shell.nix:22]: Done it. and the newlines are part of nixfmt, and get readded on reformat

**Review by fortuneteller2k — APPROVED** [review]
LGTM

#### Flakes
- [x] `nix develop` and `cargo build` works
- [x] `nix build` works

#### Channels
- [x] `nix-shell` and `cargo build` works
- [x] `nix-build` works

**Comment by legendofmiracles** [comment]
Will close #140, sorry for all the hard work on that PR :/

**Comment by elkowar** [comment]
Looks epic! thanks a ton!

**Pull request merged**